database/database_internal_test.go

```diff
@@ -228,3 +230,72 @@ func TestMultiThread(t *testing.T) {
 
 	wg.Wait()
 }
+
+func TestDownloadInfo(t *testing.T) {
+	cases := []struct {
+		downloadInfo      downloader.DownloadInfo
+		downloadInfoCount int
+	}{
+		{
+			downloadInfo: downloader.DownloadInfo{
+				Path:       "home",
+				TargetType: cloudprotocol.DownloadTargetComponent,
+			},
+			downloadInfoCount: 1,
+		},
+		{
+			downloadInfo: downloader.DownloadInfo{
+				Path:            "/path/file",
+				TargetType:      cloudprotocol.DownloadTargetLayer,
+				InterruptReason: "error",
+				Released:        true,
+			},
+			downloadInfoCount: 2,
+		},
+	}
+
+	for _, tCase := range cases {
+		if err := db.SetDownloadInfo(tCase.downloadInfo); err != nil {
+			t.Errorf("Can't set download info: %v", err)
+		}
+
+		downloadInfo, err := db.GetDownloadInfo(tCase.downloadInfo.Path)
+		if err != nil {
+			t.Errorf("Can't get download info: %v", err)
+		}
+
+		if !reflect.DeepEqual(downloadInfo, tCase.downloadInfo) {
+			t.Errorf("Unexpected download info")
+		}
+
+		downloadInfos, err := db.GetDownloadInfos()
+		if err != nil {
+			t.Errorf("Can't get list download info: %v", err)
```

Comment:
`download info list` or `list of download info`

Reply:
Done